qt/python/mantidqt/widgets/sliceviewer/presenter.py

```diff
@@ -396,10 +396,13 @@ def refresh_view(self):
         # the meantime, so check it still exists. See github issue #30406 for detail.
         if sip.isdeleted(self.view):
             return
+        ws = self.model._get_ws()
         # we don't want to use model.get_ws for the image info widget as this needs
         # extra arguments depending on workspace type.
-        self.view.data_view.image_info_widget.setWorkspace(self.model._get_ws())
```

Comment:
This is a really nice fix. Just one thing to be careful about when dealing with locks - exceptions. In you Python export methods we use the `ReleaseGlobalInterpreterLock` struct to ensure we always lock the GIL again.
In this case if anything throws an exception after the lock then the unlock won't run. In Python we can use `try...finally` to do something similar to RAII so that the `finally` clause always runs regardless if there is an exception, e.g.
 
```python
ws.readLock()
try:
    self.view.data_view.image_info_widget.setWorkspace(ws)
    self.new_plot()
finally:
    ws.unlock()
```